docs/pipelines/index.md

```diff
@@ -126,3 +126,80 @@ All of the scripts that involve LLM data generation accept a common set of param
 - **--server_args**: Any other arguments you need to pass to a corresponding server.
   E.g. use `--server_args="--gpu-memory-utilization=0.99"` to change gpu memory utilization of a
   vLLM server.
+
+## Using Config Files
+
+You can use YAML config files to pass parameters to any pipeline script. This is most applicable when using parameters that require extra escaping, such as strings with special characters.
+
+### The Problem
+
+Parameters like `end_reasoning_string='</think>'` can cause shell escaping issues:
+
+```bash
+# Problematic - angle brackets can be interpreted as shell redirection or cause Hydra parsing errors
+ns generate \
+    ... \
+    ++parse_reasoning=True \
+    ++end_reasoning_string='</think>'  # May cause errors!
+```
+
+Common error:
+```
+hydra.errors.OverrideParseException: LexerNoViableAltException: ++end_reasoning_string=\</think\>
+```
+
+### The Solution
+
+**1. Create a config file** (`/workspace/reasoning_config.yaml`):
```

Comment:
Maybe we should go with `/nemo_run/code` instead of /workspace, so that this can be packaged and uploaded? Feels like a more common use-case instead of keeping it as a standalone file. Not fully sure, but at least could mention that as an option (and add reference to packaging docs to ensure people commit this when applicable)